File: kotlin/matrix/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {

    private val m: List<List<Int>> = matrixAsString.split("\n")
        .map { r ->
            r.split(" ")
                .map { i ->
                    i.toInt()
                }
        }

    fun column(colNr: Int): List<Int> {
        return m.map { it[colNr - 1] }
    }

    fun row(rowNr: Int): List<Int> {
        return m[rowNr - 1]
    }
}

File: csharp/matrix/Matrix.cs
public class Matrix(string input)
{
    private readonly int[][] m = [.. input.Split('\n').Select(r => r.Split(' ').Select(i => int.Parse(i)).ToArray())];

    public int[] Row(int row) => m[row - 1];

    public int[] Column(int col) => [.. m.Select(r => r[col - 1])];
}